plugins/aggregators/starlark/testdata/min_max.star

```diff
@@ -0,0 +1,40 @@
+# Example of a min_max aggregator implemented with a starlark script.
+
+supported_types = (["int", "float"])
+
+def add(cache, metric):
+    aggregate = cache.get(metric)
+    if aggregate == None:
+        aggregate = {
+            "name": metric.name, 
+            "tags": metric.tags, 
+            "fields": {}
+        }
+        for k, v in metric.fields.items():
+            if type(v) in supported_types:
+                aggregate["fields"][k] = {
+				    "min": v,
+				    "max": v,
+			    }
+        cache[metric] = aggregate
+    else:
+        for k, v in metric.fields.items():
+            if type(v) in supported_types:
+                min_max = aggregate["fields"].get(k)
+                if min_max == None:
+                    aggregate["fields"][k] = {
+				        "min": v,
+				        "max": v,
+			        }
+                elif v < min_max["min"]:
+                    aggregate["fields"][k]["min"] = v
+                elif v > min_max["max"]:
+                    aggregate["fields"][k]["max"] = v
+        
+def push(cache, accumulator):
+    for a in cache:
+        fields = {}
+        for k in cache[a]["fields"]:
+            fields[k + "_min"] = cache[a]["fields"][k]["min"]
+            fields[k + "_max"] = cache[a]["fields"][k]["max"]
+        accumulator.add_fields(cache[a]["name"], fields, cache[a]["tags"])
```

Comment:
any thoughts on why we should go with an accumulator as opposed to just returning an array of metrics?

Reply:
If you check well, to test the implementation of the aggregator based on a Starlark script, I simply rewrote 2 existing aggregators which are minmax and merge with the corresponding Starlak script. And I tested them with the exact same tests. In this case, it is the minmax aggregator, and this is actually [the equivalent of the Push function](https://github.com/influxdata/telegraf/blob/master/plugins/aggregators/minmax/minmax.go#L89-L98). 

Regarding your initial question, to be honest, I don't know 😄

Reply:
In the starlark processor you just return the new metric or list of metrics from the apply function. I think it would make sense to follow the same pattern here and return the new metric from the push function instead of adding it to the accumulator. If we return the new metrics, there will be no need for the function to have an accumulator at all.

It might also make sense to rename push() to apply() to make starlark aggregators more like starlark processors.

Reply:
I second that.

Reply:
@srebhan my bad, I did it already, I just forgot to resolve it